src/bin/part1.rs

```diff
@@ -37,6 +37,19 @@ pub enum Direction {
     West,
 }
 
+impl Add<Direction> for Pos {
+    type Output = Self;
+
+    fn add(self, direction: Direction) -> Self::Output {
+        match direction {
+            Direction::North => Self::new(self.x, self.y + 1),
+            Direction::South => Self::new(self.x, self.y - 1),
+            Direction::East => Self::new(self.x + 1, self.y),
+            Direction::West => Self::new(self.x - 1, self.y),
+        }
+    }
+}
```

Comment:
This `impl` block tells the system how to interpret `pos + direction`, where `pos: Pos` and `direction: Direction`. The generic on line 40 specifies what type we're adding to `Pos`.